### PR TITLE
Return ENotCapable error when accessing unknown files on root (for #3263 and #3264)

### PR DESCRIPTION
**Pull request by ptitSeb**
Fixes for wasm.fyi tests:
 * When accessing bad path from the root, return ENOTCAPABLE error instead of ENOENT (root can only contains pre-openned folder, no other capability)
 * Accessing ROOT for other things then entering folder should result in ENOTCAPABLE

### CODE REVIEW

**Review by syrusakbary — APPROVED** [review]
We need to add also a test

**Comment by ptitSeb** [comment]
> We need to add also a test

This is a chalange here: the original tests are based on C code and around specific syscall error code return or Exit Code value. There is no infrastructure to do that on wasmer tests

File: lib/cli/src/error.rs [hunk 1]
```diff
@@ -25,8 +26,16 @@ impl PrettyError {
...
+                let trapcode = runtime.map(|e| e.clone().to_trap());
                 eprintln!("{:?}", PrettyError { error });
-                1
+                match trapcode {
+                    #[cfg(target_os = "windows")]
+                    Some(_) => 3,
+                    #[cfg(not(target_os = "windows"))]
+                    Some(_) => 128 + libc::SIGABRT,
+                    _ => 1,
+                }
```
Comment (syrusakbary) [inline lib/cli/src/error.rs:38]:
```suggestion
                let runtime: Option<&RuntimeError> = error.downcast_ref();
                let trapcode = runtime.map(|e| e.clone().to_trap());
                eprintln!("{:?}", PrettyError { error });
                match trapcode {
                    Some(err) => {
                        match err.downcast::<WasiError>() {
                            Ok(WasiError::Exit(exit_code)) => {
                                // We should exit with the provided exit code
                                std::process::abort();
                            }
                            _ => 1,
                        }
                    }
                    _ => 1,
                }
```

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**